app/controllers/users/phone_confirmation_controller.rb

```diff
@@ -0,0 +1,97 @@
+module Users
+  class PhoneConfirmationController < ApplicationController
+    before_action :authenticate_user!
+    before_action :check_for_unconfirmed_mobile
+
+    def show
+      if FeatureManagement.prefill_otp_codes?
+        @code_value = confirmation_code
+      end
+
+      @unconfirmed_mobile = unconfirmed_mobile
+      @reenter_phone_number_path = if current_user.mobile.present?
+                                     edit_user_registration_path
+                                   else
+                                     users_otp_path
+                                   end
+    end
+
+    def new
+      send_new_confirmation_code
+      redirect_to phone_confirmation_path
+    end
+
+    def confirm
```

Comment:
As discussess offline, we'll leave this for now and revisit later perhaps.  My feeling is the CRUD doesn't make a lot of sense there, but I'm new a lot of this stuff.